packages/tooltip/src/Tooltip.tsx

```diff
@@ -0,0 +1,291 @@
+  // typescript is not recognizing isControlled checks that controlledSetOpen exists
+  const setOpen = isControlled
+    ? (controlledSetOpen as Function)
+    : uncontrolledSetOpen;
+
+  const triggerRef = useRef<HTMLElement>(null);
+  const tooltipRef = useRef<HTMLDivElement>(null);
+
+  const tooltipId = useMemo(
+    () => id || `tooltip-${Math.floor(Math.random() * Math.floor(10))}`,
+    [id],
+  );
+
+  const mapToTriggerEvent = (triggerType: TriggerEvent, triggerProps?: any) => {
+    if (triggerType === TriggerEvent.Hover) {
+      return {
+        onMouseEnter: debounce(() => {
+          setOpen(!open);
+        }, 35),
+        onMouseLeave: debounce(handleClose, 35),
+        onFocus: () => setOpen(true),
+        onBlur: handleClose,
+      };
+    }
+
+    if (triggerProps.onClick) {
+      return {
+        onClick: () => {
+          triggerProps.onClick();
+          setOpen(!open);
+        },
+      };
+    }
+
+    return {
+      onClick: () => setOpen(!open),
+    };
+  };
+
+  const handleClose = () => {
+    if (!shouldClose || shouldClose()) {
+      setOpen(false);
+    }
+  };
+
+  useHandleEscape(handleClose);
+
+  const handleBackdropClick = (e: MouseEvent) => {
+    const tooltipReference = tooltipRef && tooltipRef.current;
+
+    if (
+      triggerEvent === 'click' &&
+      tooltipReference &&
+      !tooltipReference.contains(e.target as HTMLElement)
+    ) {
+      handleClose();
+    }
+  };
+
+  useEventListener('click', handleBackdropClick, { enabled: open });
+
+  const triggerRect =
+    triggerRef &&
+    triggerRef.current &&
+    triggerRef.current.getBoundingClientRect();
+
+  // We are doing this to get the final alignment and justification from Popover
+  // And to make sure we're responding to not just the desired alignment, but the alignment
+  // As a result of our calcPosition() function
+  const [alignment, setAlignment] = useState(align);
+  const [justification, setJustification] = useState(justify);
+
+  const triangleStyle = trianglePosition(
+    alignment,
+    justification,
+    triggerRect,
+  ) as { containerStyle: string; notchStyle: string };
+
+  const tooltip = (
+    <Popover
+      active={open}
+      align={align}
+      justify={justify}
+      usePortal={true}
+      adjustOnMutation={true}
+      spacing={12}
+      setAlignment={setAlignment}
+      setJustification={setJustification as any}
+    >
+      <div
+        {...rest}
+        role="tooltip"
+        id={tooltipId}
+        className={cx(className, baseStyles, tooltipVariants[variant])}
+        ref={tooltipRef}
+      >
+        <div className={triangleStyle.containerStyle}>
+          <div
+            className={cx(triangleStyle.notchStyle, notchVariants[variant])}
+          />
+        </div>
+        {children}
+      </div>
+    </Popover>
+  );
+
+  const sharedTriggerProps = {
+    ref: triggerRef,
+    'aria-describedby': tooltipId,
+  };
+
+  if (trigger) {
+    if (typeof trigger === 'function') {
```

Comment:
So....if you do `typeof class Foo {}` it returns function. Not sure if we should even worry about it.

The reason React long ago had you extend React.Component was so it could [add a flag on the prototype](https://github.com/facebook/react/blob/master/packages/react/src/ReactBaseClasses.js#L31) so it could sniff out whether it was a class for just this reason.

Reply:
You're right that using a class component breaks things. I'm trying to check if trigger is an instanceof React.Component, but that's not working for me either. Having trouble figuring out how to access that flag ^

Reply:
Fixed this ish, but when I'm testing now if I pass {Button} to the trigger prop, React complains: `Cannot call a class as a function`

Reply:
(So could be a good thing, just natively not supported, no one will try and use this way... Also supplied better direction/example in docs now re proper usage)